sql/index/pilosa/driver_test.go

```diff
@@ -349,6 +350,364 @@ func TestAscendDescendIndex(t *testing.T) {
 	}
 }
 
+func TestIntersection(t *testing.T) {
+	if !dockerIsRunning {
+		t.Skipf("Skip TestUnion: %s", dockerCmdOutput)
+	}
+	require := require.New(t)
+
+	db, table := "db_name", "table_name"
+	idxLang, expLang := "idx_lang", makeExpressions("lang")
+	idxPath, expPath := "idx_path", makeExpressions("path")
+
+	path, err := ioutil.TempDir(os.TempDir(), "indexes")
+	require.NoError(err)
+	defer os.RemoveAll(path)
+
+	d := NewDriver(path, newClientWithTimeout(200*time.Millisecond))
+	sqlIdxLang, err := d.Create(db, table, idxLang, expLang, nil)
+	require.NoError(err)
+
+	sqlIdxPath, err := d.Create(db, table, idxPath, expPath, nil)
+	require.NoError(err)
+
+	itLang := &testIndexKeyValueIter{
+		offset:      0,
+		total:       10,
+		expressions: expLang,
+		location:    offsetLocation,
+	}
+
+	itPath := &testIndexKeyValueIter{
+		offset:      0,
+		total:       10,
+		expressions: expPath,
+		location:    offsetLocation,
+	}
+
+	ctx := sql.NewContext(context.Background())
+
+	err = d.Save(ctx, sqlIdxLang, itLang)
+	require.NoError(err)
+
+	err = d.Save(ctx, sqlIdxPath, itPath)
+	require.NoError(err)
+
+	lookupLang, err := sqlIdxLang.Get(itLang.records[0].values...)
+	require.NoError(err)
+	lookupPath, err := sqlIdxPath.Get(itPath.records[itPath.total-1].values...)
+	require.NoError(err)
+
+	m, ok := lookupLang.(sql.Mergeable)
+	require.True(ok)
+	require.True(m.IsMergeable(lookupPath))
+
+	interLookup, ok := lookupLang.(sql.SetOperations)
+	require.True(ok)
+	interIt, err := interLookup.Intersection(lookupPath).Values()
+	require.NoError(err)
+	_, err = interIt.Next()
+	require.True(err == io.EOF)
+	require.NoError(interIt.Close())
+
+	lookupLang, err = sqlIdxLang.Get(itLang.records[0].values...)
+	require.NoError(err)
+	lookupPath, err = sqlIdxPath.Get(itPath.records[0].values...)
+	require.NoError(err)
+
+	interLookup, ok = lookupPath.(sql.SetOperations)
+	require.True(ok)
+	interIt, err = interLookup.Intersection(lookupLang).Values()
+	require.NoError(err)
+	loc, err := interIt.Next()
+	require.NoError(err)
+	require.True(bytes.Equal(loc, itPath.records[0].location))
+	_, err = interIt.Next()
+	require.True(err == io.EOF)
```

Comment:
maybe you can use require.Equal in that cases.

Reply:
I afraid that it won't work for byte slices, but looks that it works, so I've updated.